src/app/back/controller/BookController.php

```diff
@@ -55,17 +56,81 @@ public function all(): void
     }
 
     /**
-     * @param array $params
+     * Affiche le livre dont l'id est passé en paramètre
+     * @param int $id
+     */
+    public function one(int $id)
+    {
+        try {
+            $books[] = $this->bookManager->one($id);
+            $this->render('back/view/list.php', compact('books'));
+        } catch (Exception $e) {
+            $message = $e->getMessage();
+            echo $message;
+        }
+    }
+
+    /**
+     * Crée un livre avec les paramètres reçus
+     * Si l'id n'est pas précisé, il sera créé automatiquement
+     * @param array $params Tableau associatif dont les clefs et valeurs
+     * correspondent respectivement aux champs "name" et "value" du formulaire
      */
     public function add(array $params): void
     {
-        $book = new Book();
-        $book->setId(array_key_exists('bookId', $params) ? (int)$params['bookId'] : null);
-        $book->setName($params['bookName']);
-        $book->setPublisher($params['bookPublisher']);
-        $book->setPrice((float)$params['bookPrice']);
-        $this->bookManager->insert($book);
-        $this->all();
+        try {
+            $book = new Book();
+            $book->setId(array_key_exists('bookId', $params) ? (int)$params['bookId'] : null);
+            $book->setName($params['bookName']);
+            $book->setPublisher($params['bookPublisher']);
+            $book->setPrice((float)$params['bookPrice']);
+            $this->bookManager->insert($book);
+        } catch (Exception $e) {
+            $message = $e->getMessage();
+            echo $message;
+        }
+            // Appel de la fonction pour le message
+//        } finally {
+//            $this->all();
+//        }
```

Comment:
Ca par contre ça va poser problème

Reply:
Oui, j'ai oublié remettre le $this->all() au dessus.